File: LCP40/main.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	fmt.Println(maxmiumScore([]int{1,2,8,9}, 3))	// 18
	fmt.Println(maxmiumScore([]int{3,3,1}, 1))	// 0
	fmt.Println(maxmiumScore([]int{1, 3, 4, 5}, 4)) // 0
	fmt.Println(maxmiumScore([]int{9,5,9,1,6,10,3,4,5,1}, 2)) // 18
}

func maxmiumScore(cards []int, cnt int) int {
	sort.Slice(cards, func(i, j int) bool {
		return cards[i] > cards[j]
	})
	ans := 0
	odd := -1
	even := -1
	for i := 0; i < cnt; i++ {
		ans += cards[i]
		if cards[i] % 2 == 0 {
			even = cards[i]
		} else {
			odd = cards[i]
		}
	}

	if ans%2 == 0 {
		return ans
	}
	tmp := 0
	for i := cnt; i < len(cards); i++ {
		if cards[i] % 2 == 1 {
			if even != -1 {
				tmp = max(tmp, ans - even + cards[i])
			}
		} else {
			if odd != -1 {
				tmp = max(tmp, ans - odd + cards[i])
			}
		}
	}
	return tmp
}

File: LCP40/main.rs
struct Solution{}

impl Solution {
    pub fn maxmium_score(mut cards: Vec<i32>, cnt: i32) -> i32 {
        cards.sort_by(|a,b| b.cmp(a));
        let mut tmp = 0;
        let mut ans = 0;
        let mut odd = -1;
        let mut even = -1;
        for i in 0..cnt {
            tmp += cards[i as usize];
            if cards[i as usize] % 2 == 0 {
                even = cards[i as usize];
            } else {
                odd = cards[i as usize];
            }
        }

        if tmp % 2 == 0 {
            return tmp
        }

        for i in cnt as usize..cards.len() {
            if cards[i as usize] % 2 == 0 {
                if odd != -1 {
                    ans = std::cmp::max(ans, tmp - odd + cards[i as usize]);
                }
            } else {
                if even != -1 {
                    ans = std::cmp::max(ans, tmp - even + cards[i as usize]);
                }
            }
        }
        ans
    }
}

fn main() {
    println!("{}", Solution::maxmium_score(vec![9,5,9,1,6,10,3,4,5,1], 2))
}